### PR TITLE
#55 initial prototype of the CALM translator service

### PR DESCRIPTION
**Pull request by Budlee**
First work in progress for a Translator service
@matthewgardner, @rocketstack-matt

### CODE REVIEW

**Review by matthewgardner — APPROVED** [review]

**Comment by matthewgardner** [comment]
@rocketstack-matt lets merge this and get some of the other issues moving. :)

**Comment by Budlee** [comment]
@jpgough-ms , review of the prototype please

**Review by rocketstack-matt — COMMENTED** [review]
Is there any reason for there to be a top level translator folder then the calm-translator? Can we move everything up one level?

Would suggest it is either translator (the 'calm-' is superfluous given where it is) or the far better babel 😄

**Review by rocketstack-matt — COMMENTED** [review]
Can we add a README, doesn't need much but let's at least say this is a WIP and the purpose of it and how to run it.

**Comment by rocketstack-matt** [comment]
I've run this up and been able to generate the workspace.json then use structurizr-cli to generate to C4 in plantuml;given it's a WIP I'm ok to merge in once we have the README and remove the unnecessary directory level.

File: translator/calm-translator/src/main/java/org/finos/calmtranslator/controller/Translator.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	private final C4ModelTranslator c4ModelTranslator;
+
+	public Translator(C4ModelTranslator c4ModelTranslator) {
+		this.c4ModelTranslator = c4ModelTranslator;
+	}
+
+	@PostMapping("/c4")
+	@ResponseStatus(HttpStatus.CREATED)
+	public String c4Translation(
+			@RequestBody Core calmModel
```
Comment (jpgough-ms) [inline translator/calm-translator/src/main/java/org/finos/calmtranslator/controller/Translator.java]: Should this be `Core` or `Calm` (not sure by the way 😄 )

Comment (jpgough-ms) [reply translator/calm-translator/src/main/java/org/finos/calmtranslator/controller/Translator.java]: Oh I see now that this is generated code from the JSON schema. It's an interesting one for @rocketstack-matt around naming for model generation

**Comment by Budlee** [comment]
> I've run this up and been able to generate the workspace.json then use structurizr-cli to generate to C4 in plantuml;given it's a WIP I'm ok to merge in once we have the README and remove the unnecessary directory level.

If you take the output and put it in structurizer it has the views now generated by default
Screenshot 2024-03-13 at 19 24 48



> Can we add a README, doesn't need much but let's at least say this is a WIP and the purpose of it and how to run it.

Added this in




> Is there any reason for there to be a top level translator folder then the calm-translator? Can we move everything up one level?
> 
> Would suggest it is either translator (the 'calm-' is superfluous given where it is) or the far better babel 😄

Made this top level

**Review by jpgough-ms — APPROVED** [review]

**Comment by jpgough-ms** [comment]
Tag: @simonbrowndotje who might find this interesting 😃

File: translator/calm-translator/src/main/java/org/finos/calmtranslator/controller/Translator.java (hunk 1 above)
Comment (Budlee) [reply translator/calm-translator/src/main/java/org/finos/calmtranslator/controller/Translator.java]: Yeah, it's a bit frustrating that you can;'t override it, however, it works as it is.
You can do prefix and suffix additions, however, you can't provide a map of classes to re-name

Comment (Budlee) [reply translator/calm-translator/src/main/java/org/finos/calmtranslator/controller/Translator.java]: Yeah, it's a bit frustrating that you can;'t override it, however, it works as it is.
You can do prefix and suffix additions, however, you can't provide a map of classes to re-name

Comment (Budlee) [reply translator/calm-translator/src/main/java/org/finos/calmtranslator/controller/Translator.java]: Yeah, it's a bit frustrating that you can;'t override it, however, it works as it is.
You can do prefix and suffix additions, however, you can't provide a map of classes to re-name

**Pull request merged**